File: src/Test/ArrayOpsFFICheatcode.rs
pub fn Test_ArrayOpsFFICheatcode_runArrayOpsFFICheatcode(mut limit: i64) -> i64 {
    let mut sum = 0;
    let step = if limit >= 1 { 1 } else { -1 };
    let mut i = 1;
    loop {
        if i % 2 == 0 {
            sum += i;
        }
        if i == limit {
            break;
        }
        i += step;
    }
    sum
}

File: src/Test/ArrayOpsFFICheatcode.fs
module Test.ArrayOpsFFICheatcode

let runArrayOpsFFICheatcode (n: obj) =
    let finish = unbox<int> n
    let step = if finish >= 1 then 1 else -1
    let mutable sum = 0
    for value in 1 .. step .. finish do
        if value % 2 = 0 then sum <- sum + value
    sum :> obj
